src/machines/workflow_machines/local_acts.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{
+    machines::{local_activity_state_machine::ResolveDat, WFMachinesError},
+    protosext::HistoryEventExt,
+    worker::{ExecutingLAId, LocalActRequest, NewLocalAct},
+};
+use std::{
+    collections::{HashMap, HashSet},
+    time::SystemTime,
+};
+use temporal_sdk_core_protos::{
+    coresdk::{common::WorkflowExecution, workflow_commands::ScheduleLocalActivity},
+    temporal::api::history::v1::HistoryEvent,
+};
+
+#[derive(Default)]
+pub(super) struct LocalActivityData {
+    /// Queued local activity requests which need to be executed
+    new_requests: Vec<ScheduleLocalActivity>,
+    /// Queued cancels that need to be dispatched
+    cancel_requests: Vec<ExecutingLAId>,
+    /// Seq #s of local activities which we have sent to be executed but have not yet resolved
+    executing: HashSet<u32>,
+    /// Maps local activity sequence numbers to their resolutions as found when looking ahead at
+    /// next WFT
+    resolutions: HashMap<u32, ResolveDat>,
+}
+
+impl LocalActivityData {
+    pub(super) fn enqueue(&mut self, act: ScheduleLocalActivity) {
+        self.new_requests.push(act);
+    }
+
+    pub(super) fn enqueue_cancel(&mut self, cancel: ExecutingLAId) {
```

Comment:
I would personally like to see a cleanup PR that goes back through and makes all acronyms consistent (I like pascal case instead of all caps), decide whether to use Dat or Data, Act or Activity, LA or LocalActivity, etc.

Reply:
Yep, we can do that at some point.